### DISCUSSION TITLE
AULA CUSTOM VALIDATORS - O tipo 'null' não pode ser atribuído ao tipo 'Validators'.ts(2322)

### DISCUSSION THREAD

**Discussion by ernaneSNcarvalho** [balta-io/7187#54] [Perguntas e Respostas] [answered]
Segui a aula inclusive a custom validator ate copiei o codigo pra nao ter erro, porem aparece o erro ja citado acima, ele nao permite retornar null em tipo validator. Como resolver. ?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @ernaneSNcarvalho como vai?
Preciso de mais informações para poder te ajudar... qual arquivo deu erro, tem mais informações do erro? Algum código pra eu ver?

**Comment by dan-gama** [comment]
Também tive esse mesmo problema. Na versão do angular que estou utilizando (12.2.12), a propriedade **strictNullChecks** vem com valor **true** como padrão, impedindo de retornar valores nulos. Para resolver esse problema, abra o seu arquivo **tsconfig.json** e adicione a propriedade "strictNullChecks": false dentro de "compilerOptions"

Segue abaixo descrição da propriedade do site oficial:

> **strictNullChecks**
> When strictNullChecks is false, null and undefined are effectively ignored by the language. This can lead to unexpected errors at runtime.
> 
> When strictNullChecks is true, null and undefined have their own distinct types and you’ll get a type error if you try to use them where a concrete value is expected.

**Reply by andrebaltieri** [reply]
@ernaneSNcarvalho isto te ajuda?

**Reply by mannixbr** [reply]
> Também tive esse mesmo problema. Na versão do angular que estou utilizando (12.2.12), a propriedade **strictNullChecks** vem com valor **true** como padrão, impedindo de retornar valores nulos. Para resolver esse problema, abra o seu arquivo **tsconfig.json** e adicione a propriedade "strictNullChecks": false dentro de "compilerOptions"
> 
> Segue abaixo descrição da propriedade do site oficial:
> 
> > **strictNullChecks**
> > When strictNullChecks is false, null and undefined are effectively ignored by the language. This can lead to unexpected errors at runtime.
> > When strictNullChecks is true, null and undefined have their own distinct types and you’ll get a type error if you try to use them where a concrete value is expected.

*********************************************************************************************************************************************
Agradecido ao ghdanilo, resolveu meu problema.
